kakao/src/main/kotlin/com/agoda/kakao/screen/Screen.kt

```diff
@@ -23,16 +26,111 @@ import com.agoda.kakao.common.views.KBaseView
+    private var dataInterceptor: Interceptor<DataInteraction, ViewAssertion, ViewAction>? = null
+    private var webInterceptor: Interceptor<Web.WebInteraction<*>, WebAssertion<*>, Atom<*>>? = null
+
+    override val view: ViewInteractionDelegate = ViewInteractionDelegate(Espresso.onView(ViewMatchers.isRoot()))
+
+    private var isActive = false
 
     /**
      * The visibility of rootView will be checked when entering the screen
-     * @rootView.isVisible() Will be called after land onScreen<>()
+     * @rootView.isVisible() Will be called after land onScreen<>() and after invoking screen directly.
      */
     open var rootView: KBaseView<*>? = null
 
+    /**
+     * Sets the interceptors for the screen.
+     * Interceptors will be invoked on all interactions while the screen is active.
+     *
+     * The screen is considered `active` when it is invoked in one of the following ways:
+     * ```
+     * val screen = SomeScreen()
+     *
+     * screen { // Active
+     *     view { click() }
+     *     ...
+     * } // Inactive
+     *
+     * // OR
+     *
+     * onScreen<SomeScreen>() { // Active
+     *     view { click() }
+     *     ...
+     * } // Inactive
+     * ```
+     *
+     * If you use nesting screens, all interceptors of the screens that became active will be invoked
+     * in LIFO order (using Deque).
+     *
+     * @param configurator Configuration of the interceptors
+     *
+     * @see Interceptor
+     */
+    fun intercept(configurator: Interceptor.Configurator.() -> Unit) {
+        if (isActive) {
+            deactivate(true)
+        }
+
+        Interceptor.Configurator().apply(configurator).configure().also { (viewInterceptor, dataInterceptor, webInterceptor) ->
+            this.viewInterceptor = viewInterceptor
+            this.dataInterceptor = dataInterceptor
+            this.webInterceptor = webInterceptor
+        }
+
+        if (isActive) {
+            activate()
+        }
+    }
+
+    /**
+     * Removes the interceptors from the screen.
+     *
+     * @see intercept
+     * @see Interceptor
+     */
+    fun reset() {
+        if (isActive) {
+            deactivate(true)
+        }
+
+        viewInterceptor = null
+        dataInterceptor = null
+        webInterceptor = null
+    }
+
+    /**
+     * Operator that allows usage of DSL style
+     *
+     * @param function Tail lambda with receiver which is your screen
+     */
+    operator fun invoke(function: T.() -> Unit) {
+        activate()
+        rootView?.isVisible()
+        function.invoke(this as T)
+        deactivate()
+    }
+
+    private fun activate() {
+        viewInterceptor?.let { viewInterceptors.offerFirst(it) }
+        dataInterceptor?.let { dataInterceptors.offerFirst(it) }
+        webInterceptor?.let { webInterceptors.offerFirst(it) }
+
+        isActive = true
+    }
+
+    private fun deactivate(isReset: Boolean = false) {
```

Comment:
Some weird name of the argument ```isReset```.
I understood the mission of the argument only after attentive considering the code.
Think about to rename this argument or to add comments

Reply:
Done